### ISSUE TITLE
Recognize Frame as a web3 provider

### ISSUE THREAD

**Issue by matkam** [blocknative/onboard#460]
Onboard recognizes Metamask as a web3 provider but not [Frame](https://github.com/floating/frame-extension). Frame's web3 interface should be compatible with Metamask's. There is likely a check for Metamask that is too strict.

**Comment by floating** [comment]
Hi Onboard team!

Frame injects an [EIP-1193](https://github.com/ethereum/EIPs/blob/master/EIPS/eip-1193.md) compatible `window.ethereum` provider that can be identified via `.isFrame` and works the same as MetaMask's injected provider. 

Frame also supports native extension-less connection via RPC endpoints `http://127.0.0.1:1248` and `ws://127.0.0.1:1248`. You can use [`eth-provider`](https://github.com/floating/eth-provider) to greatly simplify this connection across browsers. [`eth-provider`](https://github.com/floating/eth-provider) is already used in projects like `web3modal`, `web3-react` and `use-wallet`. 

The example below creates an [EIP-1193](https://github.com/ethereum/EIPs/blob/master/EIPS/eip-1193.md) compatible provider, just like one injected by Frame's browser extension, except this one will connect to Frame natively from the browser without an extension.

```
import provider form 'eth-provider'
const ethereum = provider('frame')
```

We'd love to see both of these use cases supported in Onboard. Please let me know if you have any questions!

**Comment by aaronbarnardsound** [comment]
Hey @floating we are still working out how to best create a nice UX for a generic web3 provider wallet selection so that any wallet that conforms to a EIP-1193 compatible provider can be selected without needing a specific Onboard wallet module. Until then, if you would like to [implement an Onboard wallet module](https://docs.blocknative.com/onboard#custom-wallet-modules) for Frame and create a PR, we would be happy to get that merged in. For Frame it seems that it would be as simple as duplicating the MetaMask module and then swapping out the `link` and the icons. Let me know if you have any questions on that front.

We had already been looking at `eth-provider` as a replacement for our implementations that are using `web3-provider-engine` as that code base is no longer maintained and your provider implementation looks much nicer :)

**Comment by floating** [comment]
Hey @aaronbarnardsound, any updates on supporting EIP-1193? If you use `eth-provider` I'm happy to help with any integration support or features you need, though it should be pretty simple to drop in.

**Comment by floating** [comment]
I've added a wallet module to connect to Frame: natively https://github.com/blocknative/onboard/pull/494

**Comment by aaronbarnardsound** [comment]
Closed via #494

**Comment by floating** [comment]
@aaronbarnardsound I think this should be kept open until Onboard works with [EIP-1193](https://github.com/ethereum/EIPs/blob/master/EIPS/eip-1193.md) injected providers. If a user goes through the effort of injecting a provider, Onboard should be able to recognize that and display the option use it.
